py-polars/polars/lazyframe/frame.py

```diff
@@ -4453,6 +4455,11 @@ def join(
             .. note::
                 Joining on any other expressions than `col`
                 will turn off coalescing.
+        maintain_order
+            Which DataFrame row order to preserve, if any.
+            Do not rely on any observed ordering without explicitly
+            setting this parameter or your code will break in a future release.
+            Not specifying any ordering will improve performance
```

Comment:
*nit*: Should remove note about left joins preserving order above.

Should this also go through a deprecation period (during which left-joins default to `maintain_order = left`).

Should also document what the valid values for this argument are.

Reply:
I've added a test that makes sure the behaviour of no specified ordering on a left join preserves the left join order. In 2.0 we can break this compatibility.